File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
       HashMap<String, Integer> map = new HashMap<String,Integer>();
        for(String s : participant){
            map.put(s,map.getOrDefault(s,0)+1);
        }
        for(String s : completion){
            if(map.get(s)==1){
                map.remove(s);
            }
            else{
                map.replace(s,map.get(s)-1);
            }
        }
        Set<String> S = map.keySet();
        for(String s : S){
            answer += s;
        }
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <bits/stdc++.h>
using namespace std;
map<string,int> mp;
string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    for(string s : participant){
        mp[s]++;
    }
    for(string s : completion){
        mp[s]--;
    }
    for(auto it : mp){
        if(it.second != 0){
            answer += it.first;
        }
    }
    return answer;
}